backend/api/src/main/kotlin/io/tolgee/api/publicConfiguration/PublicConfigurationAssembler.kt

```diff
@@ -0,0 +1,106 @@
+package io.tolgee.api.publicConfiguration
+
+import io.tolgee.api.publicConfiguration.PublicConfigurationDTO.AuthMethodsDTO
+import io.tolgee.api.publicConfiguration.PublicConfigurationDTO.OAuthPublicConfigDTO
+import io.tolgee.api.publicConfiguration.PublicConfigurationDTO.OAuthPublicExtendsConfigDTO
+import io.tolgee.component.publicBillingConfProvider.PublicBillingConfProvider
+import io.tolgee.configuration.PlausibleDto
+import io.tolgee.configuration.tolgee.TolgeeProperties
+import io.tolgee.constants.FileStoragePath
+import io.tolgee.constants.MtServiceType
+import io.tolgee.util.VersionProvider
+import org.springframework.context.ApplicationContext
+import org.springframework.stereotype.Component
+
+@Component
+class PublicConfigurationAssembler(
+  private val properties: TolgeeProperties,
+  private val applicationContext: ApplicationContext,
+  private val publicBillingConfProvider: PublicBillingConfProvider,
+  private val versionProvider: VersionProvider,
+) {
+  fun toDto(): PublicConfigurationDTO {
+    return PublicConfigurationDTO(
+      machineTranslationServices =
+        PublicConfigurationDTO.MtServicesDTO(
+          defaultPrimaryService = getPrimaryMtService(),
+          services = getMtServices(),
+        ),
+      billing = publicBillingConfProvider(),
+      version = versionProvider.version,
+      authentication = properties.authentication.enabled,
+      screenshotsUrl = properties.fileStorageUrl + "/" + FileStoragePath.SCREENSHOTS,
+      maxUploadFileSize = properties.maxUploadFileSize,
+      clientSentryDsn = properties.sentry.clientDsn,
+      needsEmailVerification = properties.authentication.needsEmailVerification,
+      userCanCreateOrganizations = properties.authentication.userCanCreateOrganizations,
+      appName = properties.appName,
+      showVersion = properties.internal.showVersion,
+      internalControllerEnabled = properties.internal.controllerEnabled,
+      maxTranslationTextLength = properties.maxTranslationTextLength,
+      recaptchaSiteKey = properties.recaptcha.siteKey,
+      chatwootToken = properties.chatwootToken,
+      capterraTracker = properties.capterraTracker,
+      ga4Tag = properties.ga4Tag,
+      postHogApiKey = properties.postHog.apiKey,
+      postHogHost = properties.postHog.host,
+      contentDeliveryConfigured = properties.contentDelivery.publicUrlPrefix != null,
```

Comment:
Current implementation uses `cdFileStorageProvider.isServerContentDeliveryConfigured()` which has different logic to check if CD is configured. Is it correct like this?